modules/EvseV2G/v2g_ctx.cpp

```diff
@@ -479,49 +479,90 @@ void log_selected_energy_transfer_type(int selected_energy_transfer_mode) {
     }
 }
 
-void configure_parameter_set(struct iso1ServiceParameterListType* parameterSetList, int16_t parameterSetId) {
+bool add_service_to_service_list(struct v2g_context* v2g_ctx, const struct iso1ServiceType &evse_service, const int16_t* parameter_set_id, uint8_t parameter_set_id_len) {
+
+    uint8_t write_idx = 0;
+    bool service_found = false;
+
+        /* Try to find service in service list */
+        for (uint8_t idx = 0; idx < v2g_ctx->evse_v2g_data.evse_service_list_len; idx++) {
+            if (v2g_ctx->evse_v2g_data.evse_service_list[idx].ServiceID == evse_service.ServiceID) {
+                write_idx = idx;
+                service_found = true;
+                break;
+            }
+        }
+
+        if (service_found == false && 
+            (v2g_ctx->evse_v2g_data.evse_service_list_len - 1) != iso1ServiceListType_Service_ARRAY_SIZE) {
+            write_idx = v2g_ctx->evse_v2g_data.evse_service_list_len;
+            v2g_ctx->evse_v2g_data.evse_service_list_len++;
+        }
+        else if (v2g_ctx->evse_v2g_data.evse_service_list_len == iso1ServiceListType_Service_ARRAY_SIZE) {
+            dlog(DLOG_LEVEL_WARNING, "Maximum service list size reached. Unable to add service ID %u", evse_service.ServiceID);
+            return false;
```

Comment:
does the return of the function has to be bool? You do not use the return in the calling function

Reply:
not used now but maybe someone in the future ;)